packages/manager/modules/manager-pci-common/src/contexts/PCICommonContext/PCICommonContext.ts

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useMemo } from 'react';
+
+interface InternalMeta {
+  has3AZ?: boolean;
+}
+
+/**
+ * We don't export it to avoid misusage
+ *
+ * This type is only used to add completion with LSP
+ */
+type PCICommonMetaType = InternalMeta | Record<string, unknown> | undefined;
+
+/**
+ * Use {usePCICommonContextFactory} for general usage.
+ *
+ * Use this to override previously set properties
+ */
+export const PCICommonContext = createContext<PCICommonMetaType>(undefined);
+
+/**
+ * Use this to merge meta properties with previously set properties (merge is only on level 1)
+ *
+ * To override all properties use the provider directly
+ * @param meta
+ */
+export function usePCICommonContextFactory(meta?: PCICommonMetaType) {
```

Comment:
I will move this hook to the `hook` folder, to ensure global coherence of the library architecture.